### PR TITLE
Improve alluxio client local cache

### PR DESCRIPTION
**Pull request by apc999**
Introduce two changes:
(1) cleanup intermediate state when fail to add a new page
(2) make sure unchecked exception not esculated

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/cache/NoExceptionCacheManager.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import com.codahale.metrics.Counter;
+
+import java.nio.channels.ReadableByteChannel;
+
+import javax.annotation.Nullable;
+
+/**
+ * A wrapper class of CacheManager without throwing unchecked exceptions.
+ */
+public class NoExceptionCacheManager implements CacheManager {
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/NoExceptionCacheManager.java:26]: Should this catch throwable, and should we log when we catch?

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java [hunk 2]
```diff
@@ -76,8 +76,14 @@ public void put(PageId pageId, byte[] page) throws IOException {
...
       Files.createFile(p);
     }
-    try (FileOutputStream fos = new FileOutputStream(p.toFile(), false)) {
-      fos.write(page);
+    try {
+      // extra try to ensure output stream is closed
+      try (FileOutputStream fos = new FileOutputStream(p.toFile(), false)) {
+        fos.write(page);
+      }
+    } catch (IOException e) {
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java:84]: Should we catch throwable here?

Comment (apc999) [reply core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java:84]: changed to `Exception`. If `Error` happens, that's something really bad and we should escalate.

File: core/client/fs/src/main/java/alluxio/client/file/cache/NoExceptionCacheManager.java (hunk 1 above)
Comment (apc999) [reply core/client/fs/src/main/java/alluxio/client/file/cache/NoExceptionCacheManager.java:26]: add log. but IMO  `Error` should be escalated as these are not something we should recover.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.2 please.

**Pull request merged**